### PR TITLE
Drop support for ROS 2 Iron

### PR DESCRIPTION
**Pull request by sea-bass**
**Public API Changes**
None

**Description**
This PR removes support for ROS 2 Iron now that it's EOL.

Additionally, this lets us re-enable some unit tests that were previously failing on Iron.

### CODE REVIEW

**Comment by sea-bass** [comment]
@MatthijsBurgh or @bjsowa -- can you go into the repo settings and remove the Iron check as required on the `ros2` branch? Thanks!

**Comment by bjsowa** [comment]
> @MatthijsBurgh or @bjsowa -- can you go into the repo settings and remove the Iron check as required on the `ros2` branch? Thanks!

I don't have admin access to this repo so I'm unable to do this.

File: rosbridge_library/test/capabilities/test_action_capabilities.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 class TestActionCapabilities(unittest.TestCase):
     def setUp(self):
         rclpy.init()
-        self.executor = SingleThreadedExecutor()
+        self.executor = MultiThreadedExecutor()
```
Comment (bjsowa) [inline rosbridge_library/test/capabilities/test_action_capabilities.py:28]: What's the reasoning behind this change?

Comment (sea-bass) [reply rosbridge_library/test/capabilities/test_action_capabilities.py:28]: The tests that were previously skipped seem to have needed this. I was getting some flaky behavior without this for some reason.

Comment (MatthijsBurgh) [reply rosbridge_library/test/capabilities/test_action_capabilities.py:28]: I have very bad experiences with the MultiThreadedExecutor, see https://github.com/ros2/rclpy/issues/1223. So we need to make sure we use CycloneDDS to limit the issues. (Or keep using the SingleThreadedExecutor)

Comment (bjsowa) [reply rosbridge_library/test/capabilities/test_action_capabilities.py:28]: They work fine on my end with single threaded executor

Comment (sea-bass) [reply rosbridge_library/test/capabilities/test_action_capabilities.py:28]: ~Alright -- I've put it back into SingleThreaded~ EDIT: https://github.com/RobotWebTools/rosbridge_suite/pull/981#issuecomment-2551307611

That rclpy is issue is very concerning! wow

**Comment by sea-bass** [comment]
Yeah, see the test was flaky on one of the 4 runs with `SingleThreadedExecutor`.

I can either keep that test disabled, or go to `MultiThreadedExecutor`. I think for a unit test, it might be OK to use these executors, so long as we don't put it in the main source code?

**Comment by bjsowa** [comment]
> I think for a unit test, it might be OK to use these executors, so long as we don't put it in the main source code?

That's fine by me

**Review by bjsowa — APPROVED** [review]

**Comment by bjsowa** [comment]
> @MatthijsBurgh or @bjsowa -- can you go into the repo settings and remove the Iron check as required on the `ros2` branch? Thanks!

@MatthijsBurgh Could you please do it?

**Comment by MatthijsBurgh** [comment]
Done

**Review by bjsowa — APPROVED** [review]

**Pull request merged**